### PR TITLE
Search and Capture while walking to Poke Stop & Initial transferring uses Release Criteria

### PR DESCRIPTION
**Pull request by joshho**
- Search and Capture while walking to Poke Stop 
- Initial transferring uses Release Criteria (instead of hard CP comparison)

### CODE REVIEW

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -322,7 +330,7 @@ def catch_pokemon(self, pokemon):
         return_value = worker.work()
 
         if return_value == PokemonCatchWorker.BAG_FULL:
-            worker = InitialTransferWorker(self)
+            worker = TransferWorker(self)
```
Comment (solderzzc) [inline pokemongo_bot/__init__.py]: Why need change name?

**Comment by solderzzc** [comment]
This should be separated.
I think 'Initial transferring uses Release Criteria (instead of hard CP comparison)'  is good, and need README update.

File: pokemongo_bot/__init__.py (hunk 1 above)
Comment (joshho) [inline pokemongo_bot/__init__.py]: The TransferWorker also handles parts of the transfer task that happens after post-capture

File: pokemongo_bot/cell_workers/move_to_fort_worker.py [hunk 2]
```diff
@@ -41,6 +43,8 @@ def work(self):
                 while True:
                     if step_walker.step():
                         break
+                    else:
+						self.bot._work_on_cell_catch(self.position)
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/move_to_fort_worker.py]: We don't want workers calling back up to the bot like this

File: pokemongo_bot/spiral_navigator.py [hunk 3]
```diff
@@ -22,9 +22,14 @@ def __init__(self, bot):
         self.origin_lon = self.bot.position[1]
         self._step_walker = None
 
-    def take_step(self):
+    def take_stepE(self):
```
Comment (TheSavior) [inline pokemongo_bot/spiral_navigator.py]: Typo?

Comment (joshho) [inline pokemongo_bot/spiral_navigator.py]: Not too familiar with Python, couldn't overload the method for some reason.

**Comment by TheSavior** [comment]
I don't think this PR is what we want. I do think the separation of the method `_work_on_cell_catch` is a good one, but I don't think we want to flatten our logic instead of making it deeper. We definitely don't want to call work_on_cell_catch from inside a worker. 

Instead walking to pokestops should take a few steps towards the pokestop on each tick and return control to the event loop. That is the direction we went for the stepper and we should continue that direction instead of going the other way.

File: pokemongo_bot/cell_workers/move_to_fort_worker.py (hunk 2 above)
Comment (joshho) [inline pokemongo_bot/cell_workers/move_to_fort_worker.py]: What do you think if the _work_on_cell_catch logic was put away into a util of sorts?

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py [hunk 4]
```diff
@@ -82,7 +81,7 @@ def work(self):
                         
                         balls_stock = self.bot.pokeball_inventory()
                         while(True):
-
```
Comment (solderzzc) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: Code need to be clean up.

**Comment by TheSavior** [comment]
@joshho Happy to elaborate more on the previous comment if you need it. I'm going to close this PR as this approach isn't compatible with the architecture we are moving towards. Please feel free to open another PR if need be.

**Pull request closed without merging**